app/controllers/supervisor/base_controller.rb

```diff
@@ -0,0 +1,18 @@
+class Supervisor::BaseController < ApplicationController
+
+  before_action :must_be_supervisor
+  helper_method :supervised_teams
+
+  protected
+
+  def must_be_supervisor
+    unless signed_in? && current_user.roles.includes?('supervisor')
+      redirect_to root_path, alert: 'Sorry, the dashboard is for supervisors only'
+    end
+  end
+
+  def supervised_teams
+    @supervised_teams = current_user.teams.where("roles.name = 'supervisor'")
```

Comment:
It's picky and a bit of personal preference but I suggest using ActiveRecord's sql building syntax:
   
```ruby
@supervised_teams = current_user.teams.where(roles: { name: 'supervisor' })
```

It produces the almost the same SQL but without the hassle of writing any of it.